### PR TITLE
 [360] impl'd IPod#isReady that relies on all containers to be ready

### PR DESCRIPTION
**Pull request by adietish**
provides #360

### CODE REVIEW

**Comment by adietish** [comment]
fixes #360

**Comment by adietish** [comment]
@jeffmaury the new method IPod#isReady is here: https://github.com/openshift/openshift-restclient-java/pull/362/files#diff-6cdd1adf23f6100cf07d8d299ff72f53R152

**Comment by adietish** [comment]
@jeffmaury rebased to include #363

File: src/main/java/com/openshift/internal/restclient/model/Pod.java [hunk 1]
```diff
@@ -149,4 +149,17 @@ public IContainer addContainer(String name) {
...
     }
 
+    @Override
+    public boolean isReady() {
+        ModelNode node = get(POD_STATUS_CONTAINER_STATUSES);
+        if (node.getType() != ModelType.LIST) {
+            return false;
+        }
+        boolean allContainersReady = true;
+        for (ModelNode containerStatus : node.asList()) {
```
Comment (rbocanegra) [inline src/main/java/com/openshift/internal/restclient/model/Pod.java]: Hi! I was wondering if you would find useful to provide a `getContainerStatuses()` method on the Pod class that would return a `List<IContainerStatus>` and iterate over them calling `IContainerStatus` `isReady()`, `getName()`, etc...
This way you can implement this functionality and also make the user able to query the container statuses if it is found useful.

Comment (adietish) [reply src/main/java/com/openshift/internal/restclient/model/Pod.java]: @rbocanegra excellent idea! Want to PR this?

Comment (rbocanegra) [reply src/main/java/com/openshift/internal/restclient/model/Pod.java]: Sure, I would need a couple of weekends. But I will try to make a PR.

Comment (adietish) [reply src/main/java/com/openshift/internal/restclient/model/Pod.java]: @rbocanegra great, looking forward to it 👍

Comment (adietish) [reply src/main/java/com/openshift/internal/restclient/model/Pod.java]: @rbocanegra filed your suggested improvement to #379

**Comment by adietish** [comment]
[merge]

**Comment by adietish** [comment]
deployed openshift-restclient-java 7.1.1-SNAPSHOT to maven repository at repository.jboss.org

**Pull request merged**